### PR TITLE
update CRD after NSTemplateSet/NSTemplateTier Namespace type refactoring

### PR DESCRIPTION
**Pull request by xcoulon**
New version of `deploy/crds/toolchain_v1alpha1_nstemplatetier.yaml` after the `Namespace` type refactoring

Updates CRT-273

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: deploy/crds/toolchain_v1alpha1_nstemplatetier.yaml [hunk 1]
```diff
@@ -38,14 +38,104 @@ spec:
                     description: The revision of the corresponding template
                     type: string
                   template:
-                    description: Optional field. Used to specify a custom template
-                    type: string
+                    description: Used to specify a custom template
```
Comment (MatousJobanek) [inline deploy/crds/toolchain_v1alpha1_nstemplatetier.yaml]: just don't forget to update this PR with the latest changes: https://github.com/codeready-toolchain/api/pull/37/commits/40020e5d9feaec34ca65f9726fd7cc6ef836777e

Comment (xcoulon) [reply deploy/crds/toolchain_v1alpha1_nstemplatetier.yaml]: ok, let me do that now

Comment (xcoulon) [reply deploy/crds/toolchain_v1alpha1_nstemplatetier.yaml]: done in 29c11d5

**Comment by MatousJobanek** [comment]
as for the test failure - right, when you updated the API, you need to also change the logic that is using NsTemplateTier

**Comment by xcoulon** [comment]
> as for the test failure - right, when you updated the API, you need to also change the logic that is using NsTemplateTier

yes, let me take care of that

File: go.mod [hunk 2]
```diff
@@ -1,45 +1,82 @@
 module github.com/codeready-toolchain/host-operator
 
 require (
```
Comment (MatousJobanek) [inline go.mod]: @xcoulon Please run also `go mod tidy` to remove unnecessary dependencies

**Comment by xcoulon** [comment]
thanks for the review @alexeykazakov 🙌

**Pull request merged**